### PR TITLE
chore(refactor): restructure the hub to handle new Transport interface (Part 2)

### PR DESCRIPTION
**Pull request by cohix**

### CODE REVIEW

File: grav/hub.go [hunk 1]
```diff
@@ -16,64 +17,49 @@ type hub struct {
...
 			Logger:   options.Logger,
 		}
 
-		// setup messages to be sent to all active connections
-		h.pod.On(h.outgoingMessageHandler())
-
 		go func() {
-			if err := h.transport.Setup(transportOpts, h.handleIncomingConnection, h.findConnection); err != nil {
-				h.log.Error(errors.Wrap(err, "failed to Setup transport"))
+			if err := h.mesh.Setup(transportOpts, h.handleIncomingConnection); err != nil {
```
Comment (rnpridgeon) [inline grav/hub.go:61]: Should we propagating transport setup errors so the caller can decide how it wants to handle the situation?

File: grav/withdraw/signaler.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	doneChan      chan bool
+	selfWithdrawn atomic.Value
+	peerWithdrawn atomic.Value
+}
+
+// NewSignaler creates a new Signaler based on the provided context
+func NewSignaler() *Signaler {
+	w := &Signaler{
+		withdrawChan:  make(chan bool, 1),
+		doneChan:      make(chan bool, 1),
```
Comment (rnpridgeon) [inline grav/withdraw/signaler.go:20]: Will we ever signal a false value or is this just to block until the signal is sent. If it's the latter would it be more appropriate to use an empty sturct?

**Review by rnpridgeon — APPROVED** [review]
I personally would prefer pulling part1 into this PR then targeting main as opposed to merging this pr in the part 1s branch then fast-forwarding part 1 and merging this (which is what I think we'd need to do in order to get part 1 building) but I'll leave that up to you to decide.

File: grav/withdraw/signaler.go (hunk 2 above)
Comment (javorszky) [reply grav/withdraw/signaler.go:20]: agree with @rnpridgeon , if it's "wait until a signal, contents whatever" then let's use `chan struct{}`. 

See https://medium.com/@matryer/golang-advent-calendar-day-two-starting-and-stopping-things-with-a-signal-channel-f5048161018, https://dave.cheney.net/2014/03/25/the-empty-struct, and https://stackoverflow.com/questions/52035390/why-using-chan-struct-when-wait-something-done-not-chan-interface

**Pull request merged**